futures-io/src/lib.rs

```diff
@@ -354,16 +354,29 @@ if_std! {
         }
     }
 
-    impl<'a> AsyncWrite for StdIo::Cursor<&'a mut [u8]> {
-        delegate_async_write_to_stdio!();
-    }
+    impl<T: AsMut<[u8]>> AsyncWrite for StdIo::Cursor<T> {
+        fn poll_write(&mut self, _: &mut task::Context, buf: &[u8])
+            -> Poll<Result<usize>>
```

Comment:
Style :)

https://github.com/rust-lang-nursery/fmt-rfcs/blob/master/guide/items.md#function-definitions

Reply:
I see that the whole file is full of this unusual style. Ignore this then. At some point I'll just apply rustfmt to the whole project and fix such things

Reply:
`rustfmt` just removes some blank lines from this file, what do you see wrong here?

Reply:
Ah, the actual RFC doesn't allow for just splitting the return onto its own line when the args fit. I'd prefer to go with what `rustfmt` does for ease of application.

Reply:
```rust
fn poll_write(&mut self, _: &mut task::Context, buf: &[u8])
    -> Poll<Result<usize>>
{
```

should be
```rust
fn poll_write(
    &mut self,
    _: &mut task::Context,
    buf: &[u8],
) -> Poll<Result<usize>> {
```
according to the style guide.

Reply:
Oh, the `if_std!`macro call messes `rustfmt` up since it doesn't know that the internals should be normal Rust code.

Reply:
I'll fix it up anyway and rebase once #1108 is merged so this will have a decent diff.